qiskit/visualization/pulse/interpolation.py

```diff
@@ -50,7 +50,34 @@ def interp1d(time, samples, nop, kind='linear'):
 
 
 linear = partial(interp1d, kind='linear')
+linear.__doc__ = """Apply linear interpolation between sampling points.
```

Comment:
I don't think `__doc__` is supposed to be explicitly assigned to. What problem are you trying to solve here?

Reply:
This creats interpolation function by partial binding of `kind` because visualizer only receives callable. We need to prepare unique docstring for each interpolation function.

Reply:
Following our style rules, it should be documented like this:
```suggestion
"""Apply linear interpolation between sampling points.
```

But it's possible there's something complicating going on with sphinx that your way becomes necessary. Can you show me the diff between the current way and the above with the way that the docs are rendered?

Reply:
Current docstring viewed from jupyter notebook
```
Signature:      
linear(
    time:numpy.ndarray,
    samples:numpy.ndarray,
    nop:int,
    *,
    kind:str='linear',
) -> Tuple[numpy.ndarray, numpy.ndarray, numpy.ndarray]
Call signature:  linear(*args, **kwargs)
Type:            partial
String form:     functools.partial(<function interp1d at 0x7f210b56c840>, kind='linear')
File:            /usr/lib/python3.6/functools.py
Docstring:      
Apply linear interpolation between sampling points.

Args:
    time (ndarray): time.
    samples (ndarray): complex pulse envelope.
    nop (int): data points for interpolation.
Returns:
    interpolated time vector and real and imaginary part of waveform.
Class docstring:
partial(func, *args, **keywords) - new function with partial application
of the given arguments and keywords.
```

and with your suggestion
```
Signature:     
linear(
    time:numpy.ndarray,
    samples:numpy.ndarray,
    nop:int,
    *,
    kind:str='linear',
) -> Tuple[numpy.ndarray, numpy.ndarray, numpy.ndarray]
Call signature: linear(*args, **kwargs)
Type:           partial
String form:    functools.partial(<function interp1d at 0x7fd74cbfd400>, kind='linear')
File:           /usr/lib/python3.6/functools.py
Docstring:     
partial(func, *args, **keywords) - new function with partial application
of the given arguments and keywords
```

`tox -edocs` doesn't work in my local env due to import error.